Q1) Import required libraries and read the provided dataset
(youtube_dislike_dataset.csv) and retrieve top 5 and bottom 5

In [7]:
import pandas as pd 
df=pd.read_csv("youtube_dislike_dataset.csv")
print("top5 entries in the data frame ")
df.head()



top5 entries in the data frame 


video_id                                       title  \
0  --0bCF-iK2E        Jadon Sancho  Magical Skills & Goals   
1  --14w5SOEUs          Migos - Avalanche (Official Video)   
2  --40TEbZ9Is  Supporting Actress in a Comedy: 73rd Emmys   
3  --4tfbSyYDE     JO1'YOUNG (JO1 ver.)' PERFORMANCE VIDEO   
4  --DKkzWVh-E                Why Retaining Walls Collapse   

                 channel_id          channel_title         published_at  \
0  UC6UL29enLNe4mqwTfAyeNuw             Bundesliga  2021-07-01 10:00:00   
1  UCGIelM2Dj3zza3xyV3pL3WQ              MigosVEVO  2021-06-10 16:00:00   
2  UClBKH8yZRcM4AsRjDVEdjMg     Television Academy  2021-09-20 01:03:32   
3  UCsmXiDP8S40uBeJYxvyulmA                    JO1  2021-03-03 10:00:17   
4  UCMOqf8ab-42UUQIdVoKwjlQ  Practical Engineering  2021-12-07 13:00:00   

   view_count   likes  dislikes  comment_count  \
0     1048888   19515       226           1319   
1    15352638  359277      7479          18729   
2      925281   11212       401            831   
3     2641597   39131       441           3745   
4      715724   32887       367           1067   

                                                tags  \
0  football soccer ftbol alemn Bundesliga season ...   
1  Migos Avalanche Quality Control Music/Motown R...   
2                                                      
3  PRODUCE101JAPAN              JO1   TheSTAR STA...   
4  retaining wall New Jersey highway Direct Conne...   

                                         description  \
0  Enjoy the best skills and goals from Jadon San...   
1  Watch the the official video for Migos - "Aval...   
2  Hannah Waddingham wins the Emmy for Supporting...   
3  JO1'YOUNG (JO1 ver.)' PERFORMANCE VIDEO\n\n---...   
4  One of the most important (and innocuous) part...   

                                            comments  
0  Respect to Dortmund fans,must be sad losing hi...  
1  Migos just makes me want to live my live to th...  
2  Hannah's energy bursts through any screen. Wel...  
3  youngVer><  REN is really PERFECT. It's not ju...  
4   Keep up with all my projects here: https://pr...

In [8]:
print("last5 entries in the data frame ")
df.tail()

last5 entries in the data frame 


video_id                                              title  \
37417  zzd4ydafGR0  Lil Tjay - Calling My Phone (feat. 6LACK) [Off...   
37418  zziBybeSAtw  PELICANS at LAKERS | FULL GAME HIGHLIGHTS | Ja...   
37419  zzk09ESX7e0                 [MV]  (MAMAMOO) - Where Are We Now   
37420  zzmQEb0Em5I              FELLIPE ESCUDERO- Master Podcast  #12   
37421  zzxPZwaA-8w  Gareth Bale brace secures dramatic comeback on...   

                     channel_id      channel_title         published_at  \
37417  UCEB4a5o_6KfjxHwNMnmj54Q           Lil Tjay  2021-02-12 05:03:49   
37418  UCWJ2lWNubArHWmf3FIHbfcQ                NBA  2021-01-16 05:39:05   
37419  UCuhAUMLzJxlP1W7mEk0_6lA            MAMAMOO  2021-06-02 09:00:10   
37420  UC8NjnNWMsRqq11NYvHAQb1g     Master Podcast  2020-10-20 20:59:30   
37421  UCEg25rdRZXg32iwai6N6l0w  Tottenham Hotspur  2021-05-23 21:00:31   

       view_count    likes  dislikes  comment_count  \
37417   120408275  2180780     35871          81360   
37418     2841917    20759      1049           2624   
37419    13346678   720854      4426          90616   
37420      252057    19198      1234           1471   
37421     2252090    34063       868           2004   

                                                    tags  \
37417  Lil Tjay Steady Calling My Phone Calling My Ph...   
37418  NBA G League Basketball game-0022000187 Lakers...   
37419  MAMAMOO  WAW  WAW MAMAMOO WAW Where Are We Now...   
37420  master masterpodcast lord lord vinheteiro z z ...   
37421  Spurs Tottenham Hotspur   Tottenham Leicester ...   

                                             description  \
37417  Official video for "Calling My Phone" by Lil T...   
37418  PELICANS at LAKERS | FULL GAME HIGHLIGHTS | Ja...   
37419  [MV]  (MAMAMOO) - Where Are We Now\n\nInstagra...   
37420  DOCTOR HAIR\nhttps://www.thedoctorhair.com/?fb...   
37421  Two minute highlights from Tottenham Hotspur's...   

                                                comments  
37417  'DESTINED 2 WIN' OUT NOW !! https://liltjay.ln...  
37418  Montrezl Harrell is going crazy with the rebou...  
37419  I honestly do not know why this song hit so ha...  
37420  Foi um prazer passar esta tarde com vocs debat...  
37421  Thank you Kane for everything you have given t...

Q2) Check the info of the data frame and write your inferences on
datatypes and the shape of the dataset

In [9]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 37422 entries, 0 to 37421
Data columns (total 12 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   video_id       37422 non-null  object
 1   title          37422 non-null  object
 2   channel_id     37422 non-null  object
 3   channel_title  37422 non-null  object
 4   published_at   37422 non-null  object
 5   view_count     37422 non-null  int64 
 6   likes          37422 non-null  int64 
 7   dislikes       37422 non-null  int64 
 8   comment_count  37422 non-null  int64 
 9   tags           37422 non-null  object
 10  description    37422 non-null  object
 11  comments       37264 non-null  object
dtypes: int64(4), object(8)
memory usage: 3.4+ MB


In [10]:
df.shape

(37422, 12)

solution for question 2

* There are 12 columns and 37422 rows with in the given data,
* Columns view_count, likes,dislikes,comment_count has integer data type and rest 8 columns having object data type
* We can consider column video_id as the unique identifer for the rows with the data
  

In [14]:
pd.isnull(df).sum()

video_id           0
title              0
channel_id         0
channel_title      0
published_at       0
view_count         0
likes              0
dislikes           0
comment_count      0
tags               0
description        0
comments         158
dtype: int64

In [17]:
new_df=df.dropna()
new_df.isnull().sum()

video_id         0
title            0
channel_id       0
channel_title    0
published_at     0
view_count       0
likes            0
dislikes         0
comment_count    0
tags             0
description      0
comments         0
dtype: int64

Q4) Check the statistical summary of the dataset.

In [18]:
new_df.describe()

view_count         likes      dislikes  comment_count
count  3.726400e+04  3.726400e+04  3.726400e+04   3.726400e+04
mean   5.710821e+06  1.672867e+05  4.996434e+03   9.966953e+03
std    2.431304e+07  5.384931e+05  3.075194e+04   1.173465e+05
min    2.036800e+04  2.200000e+01  3.000000e+00   1.000000e+00
25%    5.141305e+05  1.331725e+04  2.827500e+02   9.140000e+02
50%    1.321860e+06  4.255600e+04  7.980000e+02   2.347000e+03
75%    3.674527e+06  1.309665e+05  2.466250e+03   6.212000e+03
max    1.322797e+09  3.183768e+07  2.397733e+06   1.607103e+07

In [25]:
new_df.head()

video_id                                       title  \
0  --0bCF-iK2E        Jadon Sancho  Magical Skills & Goals   
1  --14w5SOEUs          Migos - Avalanche (Official Video)   
2  --40TEbZ9Is  Supporting Actress in a Comedy: 73rd Emmys   
3  --4tfbSyYDE     JO1'YOUNG (JO1 ver.)' PERFORMANCE VIDEO   
4  --DKkzWVh-E                Why Retaining Walls Collapse   

                 channel_id          channel_title        published_at  \
0  UC6UL29enLNe4mqwTfAyeNuw             Bundesliga 2021-07-01 10:00:00   
1  UCGIelM2Dj3zza3xyV3pL3WQ              MigosVEVO 2021-06-10 16:00:00   
2  UClBKH8yZRcM4AsRjDVEdjMg     Television Academy 2021-09-20 01:03:32   
3  UCsmXiDP8S40uBeJYxvyulmA                    JO1 2021-03-03 10:00:17   
4  UCMOqf8ab-42UUQIdVoKwjlQ  Practical Engineering 2021-12-07 13:00:00   

   view_count   likes  dislikes  comment_count  \
0     1048888   19515       226           1319   
1    15352638  359277      7479          18729   
2      925281   11212       401            831   
3     2641597   39131       441           3745   
4      715724   32887       367           1067   

                                                tags  \
0  football soccer ftbol alemn Bundesliga season ...   
1  Migos Avalanche Quality Control Music/Motown R...   
2                                                      
3  PRODUCE101JAPAN              JO1   TheSTAR STA...   
4  retaining wall New Jersey highway Direct Conne...   

                                         description  \
0  Enjoy the best skills and goals from Jadon San...   
1  Watch the the official video for Migos - "Aval...   
2  Hannah Waddingham wins the Emmy for Supporting...   
3  JO1'YOUNG (JO1 ver.)' PERFORMANCE VIDEO\n\n---...   
4  One of the most important (and innocuous) part...   

                                            comments  
0  Respect to Dortmund fans,must be sad losing hi...  
1  Migos just makes me want to live my live to th...  
2  Hannah's energy bursts through any screen. Wel...  
3  youngVer><  REN is really PERFECT. It's not ju...  
4   Keep up with all my projects here: https://pr...

In [30]:
print(type(new_df))

<class 'pandas.core.frame.DataFrame'>


In [57]:
df1=pd.DataFrame(new_df)

Q5) Convert data type of column published_at from object to pandas
DateTime. 

In [39]:
df1['published_at'] = pd.to_datetime(df1['published_at'])

Q6) Create a column as 'published_month' using the column published_at. 

In [40]:
df1['published_month'] = df1['published_at'].dt.month

Q7) Replace the numbers in the column published_month as names of the
months i.e. 1 as 'Jan', 2 as 'Feb', and so on..... 

In [59]:
df1['published_month'] = df1['published_month'].replace(month_names)

In [60]:
df1['published_month'].head()

0    Jul
1    Jun
2    Sep
3    Mar
4    Dec
Name: published_month, dtype: object

In [61]:
df1

video_id                                              title  \
0      --0bCF-iK2E               Jadon Sancho  Magical Skills & Goals   
1      --14w5SOEUs                 Migos - Avalanche (Official Video)   
2      --40TEbZ9Is         Supporting Actress in a Comedy: 73rd Emmys   
3      --4tfbSyYDE            JO1'YOUNG (JO1 ver.)' PERFORMANCE VIDEO   
4      --DKkzWVh-E                       Why Retaining Walls Collapse   
...            ...                                                ...   
37417  zzd4ydafGR0  Lil Tjay - Calling My Phone (feat. 6LACK) [Off...   
37418  zziBybeSAtw  PELICANS at LAKERS | FULL GAME HIGHLIGHTS | Ja...   
37419  zzk09ESX7e0                 [MV]  (MAMAMOO) - Where Are We Now   
37420  zzmQEb0Em5I              FELLIPE ESCUDERO- Master Podcast  #12   
37421  zzxPZwaA-8w  Gareth Bale brace secures dramatic comeback on...   

                     channel_id          channel_title        published_at  \
0      UC6UL29enLNe4mqwTfAyeNuw             Bundesliga 2021-07-01 10:00:00   
1      UCGIelM2Dj3zza3xyV3pL3WQ              MigosVEVO 2021-06-10 16:00:00   
2      UClBKH8yZRcM4AsRjDVEdjMg     Television Academy 2021-09-20 01:03:32   
3      UCsmXiDP8S40uBeJYxvyulmA                    JO1 2021-03-03 10:00:17   
4      UCMOqf8ab-42UUQIdVoKwjlQ  Practical Engineering 2021-12-07 13:00:00   
...                         ...                    ...                 ...   
37417  UCEB4a5o_6KfjxHwNMnmj54Q               Lil Tjay 2021-02-12 05:03:49   
37418  UCWJ2lWNubArHWmf3FIHbfcQ                    NBA 2021-01-16 05:39:05   
37419  UCuhAUMLzJxlP1W7mEk0_6lA                MAMAMOO 2021-06-02 09:00:10   
37420  UC8NjnNWMsRqq11NYvHAQb1g         Master Podcast 2020-10-20 20:59:30   
37421  UCEg25rdRZXg32iwai6N6l0w      Tottenham Hotspur 2021-05-23 21:00:31   

       view_count    likes  dislikes  comment_count  \
0         1048888    19515       226           1319   
1        15352638   359277      7479          18729   
2          925281    11212       401            831   
3         2641597    39131       441           3745   
4          715724    32887       367           1067   
...           ...      ...       ...            ...   
37417   120408275  2180780     35871          81360   
37418     2841917    20759      1049           2624   
37419    13346678   720854      4426          90616   
37420      252057    19198      1234           1471   
37421     2252090    34063       868           2004   

                                                    tags  \
0      football soccer ftbol alemn Bundesliga season ...   
1      Migos Avalanche Quality Control Music/Motown R...   
2                                                          
3      PRODUCE101JAPAN              JO1   TheSTAR STA...   
4      retaining wall New Jersey highway Direct Conne...   
...                                                  ...   
37417  Lil Tjay Steady Calling My Phone Calling My Ph...   
37418  NBA G League Basketball game-0022000187 Lakers...   
37419  MAMAMOO  WAW  WAW MAMAMOO WAW Where Are We Now...   
37420  master masterpodcast lord lord vinheteiro z z ...   
37421  Spurs Tottenham Hotspur   Tottenham Leicester ...   

                                             description  \
0      Enjoy the best skills and goals from Jadon San...   
1      Watch the the official video for Migos - "Aval...   
2      Hannah Waddingham wins the Emmy for Supporting...   
3      JO1'YOUNG (JO1 ver.)' PERFORMANCE VIDEO\n\n---...   
4      One of the most important (and innocuous) part...   
...                                                  ...   
37417  Official video for "Calling My Phone" by Lil T...   
37418  PELICANS at LAKERS | FULL GAME HIGHLIGHTS | Ja...   
37419  [MV]  (MAMAMOO) - Where Are We Now\n\nInstagra...   
37420  DOCTOR HAIR\nhttps://www.thedoctorhair.com/?fb...   
37421  Two minute highlights from Tottenham Hotspur's...   

                                                comments published_month  
0      Respect to 

Q8) Find the minimum and maximum value of the view_count, likes,
dislikes, and comment_count and write your inferences. 

In [66]:
print("minimum view count = ",df1["view_count"].min(), "maximum view count =",df1["view_count"].max())
print("minimum likes = ",df1["likes"].min(), "maximum likes =",df1["likes"].max())
print("minimum dislikes = ",df1["dislikes"].min(), "maximum dislikes=",df1["dislikes"].max())
print("minimum comment_count = ",df1["comment_count"].min(), "maximum comment_count =",df1["comment_count"].max())


minimum view count =  20368 maximum view count = 1322796924
minimum likes =  22 maximum likes = 31837675
minimum dislikes =  3 maximum dislikes= 2397733
minimum comment_count =  1 maximum comment_count = 16071029


We can see that the mimimum view_count is 20368 and max is 1322796924.
The mimimum likes is 22 and max is 31837675.
The mimimum dislikes is 3 and max is 2397733.
The mimimum comment_count is 1 and max is 16071029.

Q9) Find the count of unique video_id, channel_id, and channel_title.

In [73]:
count_video_id=df1["video_id"].nunique()
count_channel_id=df1["channel_id"].nunique()
count_channel_title=df1["channel_title"].nunique()
print("unique video id count",count_video_id)
print("unique chaneel id ",count_channel_id)
print("unique channel title ",count_channel_title)

unique video id count 37264
unique chaneel id  10891
unique channel title  10813


Q10) Find the top10 channel names having the highest number of videos in
the dataset and the bottom10 having the lowest number of videos. 

In [76]:
df1.groupby('channel_title')['video_id'].count().sort_values(ascending=True).head(10)

channel_title
   SilverName                   1
Miniatur Wunderland             1
MiniBloxia                      1
Mini Muka                       1
Mini Ladd                       1
MindYourLogic                   1
Mind Body Tonic With Dr Sita    1
Mimi Ar                         1
Millyz                          1
Milkair                         1
Name: video_id, dtype: int64

In [77]:
df1.groupby('channel_title')['video_id'].count().sort_values(ascending=False).head(10)

channel_title
Sky Sports Football    533
The United Stand       301
BT Sport               246
NBA                    209
NFL                    162
WWE                    122
SSSniperWolf            99
SSundee                 98
FORMULA 1               87
NHL                     86
Name: video_id, dtype: int64

Q11) Find the title of the video which has the maximum number of likes
and the title of the video has minimum likes and write your inferences

In [81]:
import numpy as np
pd.pivot_table(df1,index=['title'],values=['likes'],aggfunc=np.max).sort_values(by='likes',ascending=False).head(1)

C:\Users\abhis\AppData\Local\Temp\ipykernel_9088\739854494.py:2: FutureWarning: The provided callable <function max at 0x0000018F3803B6A0> is currently using DataFrameGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  pd.pivot_table(df1,index=['title'],values=['likes'],aggfunc=np.max).sort_values(by='likes',ascending=False).head(1)


likes
title                                  
BTS () 'Dynamite' Official MV  31837675

Find the title of the video which has the maximum number of dislikes
and the title of the video having minimum dislikes and write your
inferences.


In [82]:
pd.pivot_table(df1,index=['title'],values=['dislikes'],aggfunc=np.max).sort_values(by='dislikes',ascending=False).head(1)

C:\Users\abhis\AppData\Local\Temp\ipykernel_9088\1183259175.py:1: FutureWarning: The provided callable <function max at 0x0000018F3803B6A0> is currently using DataFrameGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  pd.pivot_table(df1,index=['title'],values=['dislikes'],aggfunc=np.max).sort_values(by='dislikes',ascending=False).head(1)


dislikes
title                                        
Cuties | Official Trailer | Netflix   2397733